sample-apps/README.md

```diff
@@ -38,3 +38,245 @@ List of sample apps across all repositories in [aws-observability](https://githu
+RandomTotalHeapSizeUpperBound: 100    # Metric - UpperBound for TotalHeapSize for random metric value every TimeInterval
+RandomThreadsActiveUpperBound: 10     # Metric - UpperBound for ThreadsActive for random metric value every TimeInterval
+RandomCpuUsageUpperBound: 100         # Metric - UppperBound for CpuUsage for random metric value every TimeInterval
+SampleAppPorts: []                    # Sampleapp ports to make calls to   
+ResourceDetector: ''                  # String to specify resource detector
+```
+
+Every sample app will assume a configuration file with these variable names.
+The testing framework MUST not test against these Metrics due to upstream unstable SDK & API in certain languages. These are mainly for being able to demonstrate some type of metrics being generated in languages that support it. 
+
+#### Metrics
+
+For Metrics, each sample app must include the 6 different types of instruments to help showcase how each instrument can be utilized in varying contexts when gathering metrics. Each sample app must implement every instrument in our included languages and keep variable names, units, descriptions, and outputs consistent across all of them. The request based metrics will also be generated alongside traces from incoming requests from invocations or incoming traffic. Any languages that are not stable in metrics or are unable to be implemented under this structure must include documentation on that issue in their respective **Documentation** document as mentioned in the **Documentation requirements**.
+
+**Request based:**
+
+AsyncCounter
+
+* Name: totalApiRequests
+* Unit: “1”
+* Description: “Increments by one every time a sampleapp endpoint is used”
+
+Counter
+
+* Name: totalBytesSent
+* Unit: “By”
+* Description: “Keeps a sum of the total amount of bytes sent while the application is alive”
+* Includes Callback Function
+
+Histogram
+
+* Name: latencyTime
+* Unit: “ms”
+* Description: “Measures latency time in buckets of 100 300 and 500”
+
+**Common Attributes for Request based metrics**
+
+These are Key Value pairs to be added on metrics.
+```
+{
+“signal”:     (string)“metric”
+“language”:   (string)<name of language used>
+“metricType”: (string)“request”
+}
+```
+
+**Random:**
+
+Counter
+
+* Name: timeAlive
+* Unit: “ms”
+* Description: “Total amount of time that the application has been alive”
+
+Asynchronous UpDown Counter
+
+* Name: totalHeapSize
+* Unit: “By”
+* Description: “The current total heap size”
+* Includes Callback Function
+
+UpDown Counter
+
+* Name: threadsActive
+* Unit: “1”
+* Description: “The total number of threads active”
+
+Asynchronous Gauge
+
+* Name: cpuUsage
+* Unit: “1”
+* Description: “Cpu usage percent”
+* Includes Callback Function
+
+**Common Attributes for Random based metrics**
+
+These are Key Value pairs to be added on metrics.
+```
+{
+“signal”:     (string)“metric”
+“language”:   (string)<name of language used>
+“metricType”: (string)“random”
+}
+```
+
+#### Logs
```

Comment:
Should we include this?

Reply:
I don't think it could hurt to include this as we do note that logs are optional amongst other warnings.  After logs are stabilized, it will be easier to just edit this section in my opinion opposed to adding from scratch.